src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -357,6 +357,113 @@ public bool IsHomeSpace
+                    var outputsProperty = node.Children<JProperty>().FirstOrDefault(x => x.Name == "Outputs");
+                    var outputs = (JArray)outputsProperty.Value;
+                    int outputIndex = 1;
+
+                    foreach (JObject output in outputs)
+                    {
+                        var outputProperties = output.Children<JProperty>();
+                        JProperty outputId = outputProperties.FirstOrDefault(x => x.Name == "Id");
+                        outputIds.Add(outputId.Value.ToString());
+
+                        var connectorsProperty = jsonWorkspace["Connectors"].Where(t => t.Value<string>("Start") == outputId.Value.ToString());
+
+                        foreach (var connector in connectorsProperty)
+                        {
+                            var connectorProperties = connector.Children<JProperty>();
+                            JProperty endProperty = connectorProperties.FirstOrDefault(x => x.Name == "End");
+                            string inputId = (String)endProperty.Value;
+
+                            var outputConnectedNode = GetNodeByInputId(inputId, jsonWorkspace);
+                            nodeInfoConnections.Add(nodeId + "|[" + outputIndex.ToString() + "|" + outputConnectedNode.Item1 + "|" + outputConnectedNode.Item2.ToString() + "]");
+                        }
+                        outputIndex++;
+                    }
+                }
+                return nodeInfoConnections.Count > 0 ? string.Join(",", nodeInfoConnections) : string.Empty;
+            }            
+        }
+
+        Tuple<string,int> GetNodeByInputId(string inputId, JObject jsonWorkspace)
+        {
+            var nodes = jsonWorkspace["Nodes"];
+
+            string nodeId = string.Empty;
+            int connectedInputIndex = 1;
+            bool foundNode = false;
+
+            foreach (var node in nodes)
+            {
+                if (!foundNode)
+                {
+                    nodeId = string.Empty;
+                    connectedInputIndex = 1;
+
+                    var nodeProperties = node.Children<JProperty>();
+                    JProperty nodeProperty = nodeProperties.FirstOrDefault(x => x.Name == "Id");
+                    nodeId = (String)nodeProperty.Value;
+
+                    JProperty nodeInputs = nodeProperties.FirstOrDefault(x => x.Name == "Inputs");
+                    var inputs = (JArray)nodeInputs.Value;
+
+                    foreach (JObject input in inputs)
+                    {
+                        var inputProperties = input.Children<JProperty>();
+                        JProperty connectedNodeInputId = inputProperties.FirstOrDefault(x => x.Name == "Id");
+
+                        if ((String)connectedNodeInputId.Value == inputId)
+                        {
+                            string hh = nodeId + "___" + connectedInputIndex.ToString();
```

Comment:
What is `hh` here? and why are we using "___"?

Reply:
> What is `hh` here? and why are we using "___"?

it was intentionally for trace some info, removed.